### DISCUSSION TITLE
MediaElement and MediaPlayerElement Roadmap?

### DISCUSSION THREAD

**Discussion by mjfusa** [microsoft/WindowsAppSDK#817] [Q&A] [answered]
Is it noted [here ](https://docs.microsoft.com/en-us/windows/apps/winui/winui3/release-notes/winui3-project-reunion-0.5-preview#missing-platform-features) that **MediaElement** and **MediaPlayerElement** are currently missing in Project Reunion. Can you share when we can expect to see these controls supported? In the meantime, are there any suggested alternatives for playing back .MP3 and Video content? Thanks!

**Comment by mrlacey** [comment]
The [WinUI Roadmap](https://github.com/microsoft/microsoft-ui-xaml/blob/main/docs/roadmap.md#winui-30-feature-roadmap) suggest these won't be before the 1.0 release and even then doesn't commit to a timeframe. (The history of that roadmap has always had this feature marked with the equivalent of "at some unknown point in the future." i.e. don't expect it any time soon.)

The only workaround I know is to play through an embedded browser control but is far from ideal. Proper media element support stopped a project I was working on from using WinUI3 and we used WPF instead. Admittedly, this is a more extreme workaround but avoided lots of issues with current gaps in WinUI3.

**Comment by castorix** [comment]
There are many other ways to play Videos
I use WMP control in a Win32 ATL child control and it works fine
But MediaElement is more powerful, like for rtsp for example

**Comment by andrewleader** [comment, marked as answer]
Hi @mjfusa, Media controls aren't part of 1.0 (which just shipped yesterday).

You can tell us how important Media controls are for your app by voting on the Media controls feature here: https://portal.productboard.com/winappsdk/1-windows-app-sdk/c/30-media-controls

Media controls currently aren't part of our 1.1 plans either, but we're still planning 1.1. We're trying to focus on some critical scenarios in 1.1 though and it's looking like media controls will be unlikely to be part of that, but it's certainly our top feature request overall.

**Reply by mrlacey** [reply]
> Media controls currently aren't part of our 1.1 plans either, but we're still planning 1.1. We're trying to focus on some critical scenarios in 1.1 though and it's looking like media controls will be unlikely to be part of that, but it's certainly our top feature request overall.

@andrewleader can you talk more about how priorities are determined, what the critical scenarios are, and what makes the media controls so hard to add. It seems odd that the top feature request isn't included.

**Reply by andrewleader** [reply]
@mrlacey I think Ryan and Bridget touched on it a bit in the community call yesterday. Our customers for 2022 include some very big 1st and 3rd party apps along with the general developer community, and we are currently weighting our 2022 prioritizing toward enabling those big 1st party apps and select 3rd party apps in 2022. Many developers have told us that they want to see Microsoft using WinAppSDK and WinUI 3 in their own apps so they can trust they can rely on it. We can't share what those apps are yet, but as we start building that out, we're hoping we can share that soon.

Those partners currently don't need Media controls in 1.1, and we have limited resources and a LOT of work to do (prep for XAML Islands, self-contained support, core infrastructure work so that built-in Windows components can use WinAppSDK, performance improvements, etc). We're trying to prioritize what's absolutely required in 1.1 for those scenarios. Not including media controls in 1.1 has been a really difficult discussion (not a final decision yet), as we know it is our top request from developers overall. We haven't planned 1.2 yet, but if I had to guess, I would guess media controls would come in 1.2 (not a promise or a plan, just a guess).
